src/runtime/hexagon/target/hexagon_device_target.cc

```diff
@@ -0,0 +1,525 @@
+  tvm_hexagon_remote_handle_t module_pointer_ = AEE_EUNKNOWN;
+  uint64_t count_channel_open_ = 0;
+  // Global lock, used for all critical sections. This can be refined
+  // in the future.
+  mutable std::mutex crit_section_;
+};
+
+std::shared_ptr<Device> CreateHexagonTarget() {
+  return std::make_shared<HexagonTarget>();
+}
+
+std::pair<void*, size_t> HexagonTarget::AddAddrMapping(const void* dsp_addr,
+                                                       void* apps_addr,
+                                                       size_t size) {
+  crit_section_.lock();
+  auto p = dsp_to_apps_.insert({dsp_addr, {apps_addr, size}});
+  crit_section_.unlock();
+  if (!p.second) {
+    TVM_LOGE_HT(
+        "failed to insert address mapping: dsp:%p -> apps:%p, size:%zu",
+        dsp_addr, apps_addr, size);
+    return std::make_pair(nullptr, 0);
+  }
+  TVM_LOGD_HT("added address mapping: dsp:%p -> apps:%p, size:%zu", dsp_addr,
+              apps_addr, size);
+  return p.first->second;
+}
+
+void HexagonTarget::RemoveAddrMapping(const void* dsp_addr) {
+  crit_section_.lock();
+  auto f = dsp_to_apps_.find(dsp_addr);
+  if (f == dsp_to_apps_.end()) {
+    TVM_LOGE_HT("failed to remove address mapping for dsp:%p", dsp_addr);
+    crit_section_.unlock();
+    return;
+  }
+  dsp_to_apps_.erase(f);
+  crit_section_.unlock();
+}
+
+std::pair<void*, size_t> HexagonTarget::GetAppsAddr(const void* dsp_addr,
+                                                    bool exact) const {
+  struct AutoUnlock {
+    explicit AutoUnlock(std::mutex& m) : m(m) {}
+    ~AutoUnlock() { m.unlock(); }
+    std::mutex& m;
+  };
+
+  crit_section_.lock();
+  AutoUnlock u(crit_section_);
+
+  // If the address is in the map, simply return the result.
+  auto f = dsp_to_apps_.find(dsp_addr);
+  if (f != dsp_to_apps_.end()) return f->second;
+  // If exact mapping is requested, then it hasn't been found.
+  if (exact) return std::make_pair(nullptr, 0);
+
+  // If the address is not in the map, maybe it points to somewhere in the
+  // interior of a mapped buffer.
+  uintptr_t dsp_v = reinterpret_cast<uintptr_t>(dsp_addr);
+  for (const auto& v : dsp_to_apps_) {
+    uintptr_t dsp_k = reinterpret_cast<uintptr_t>(v.first);
+    size_t size = v.second.second;
+    if (dsp_v >= dsp_k && dsp_v < dsp_k + size) {
+      uintptr_t apps_k = reinterpret_cast<uintptr_t>(v.second.first);
+      size_t offset = dsp_v - dsp_k;
+      uintptr_t apps_v = apps_k + offset;
+      return std::make_pair(reinterpret_cast<void*>(apps_v), size - offset);
+    }
+  }
+  TVM_LOGE_HT("failed to locate apps address for dsp:%p", dsp_addr);
+  return std::make_pair(nullptr, 0);
+}
+
+int HexagonTarget::OpenDomainChannel(bool use_unsigned_pd) {
+  if (domain_channel_handle_ != AEE_EUNKNOWN) return AEE_SUCCESS;
+
+  const DspRpcAPI* dsp_api = DspRpcAPI::Global();
+  const StubAPI* stub_api = StubAPI::Global();
+
+  stub_api->rpcmem_init_ptr()();
```

Comment:
I maybe forget a little bit, because I do it last year. However, I remember `rpcmem_init_ptr` will have problem when we set `use_unsigned_pd` be false. In your code logic of this pr, you will always set `use_unsigned_pd` be true. Could you double check it? I set it be false, because I find when we set it be `true`, we can not do parallel on DSP. But I don't have device now and can not verify it sadly.

Reply:
I can set it to false, would you prefer that?  I don't think there is a way to make it a parameter that the user can provide at run time.

Reply:
Yes. I prefer setting it to be false.

Reply:
Done.